pkg/jobs/hive/hive.go

```diff
@@ -329,8 +333,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
+	}
+
+	if err := validateEUSUpgradeVersion(client, clusterName, curator, isInterVersion); err != nil {
+		return err
+	}
+
+	klog.V(2).Info("Check if managedclusterview exists " + clusterName)
+
+	// For OCP/Kubenetes versions that removed APIs, need to acknowledge this
+	// before upgrading or else upgrade will be blocked.
+	ocpConfigMCV := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+			Labels: map[string]string{
+				MCVUpgradeLabel: clusterName,
+			},
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Kind:      "ConfigMap",
+				Name:      "admin-acks",
+				Namespace: "openshift-config",
+				Version:   "v1",
+			},
+		},
+	}
+
+	ocpConfigView := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      clusterName + "admack",
+		Namespace: clusterName,
+	}, &ocpConfigView); err != nil {
+		// check if mcv exists before creating
+
+		klog.V(2).Info("Create managedclusterview " + clusterName + "admack")
+		if err := client.Create(context.TODO(), ocpConfigMCV); err != nil {
+			return err
+		}
+	}
+
+	ocpConfigGetErr := errors.New("Failed to get remote admin-ack configmap")
+	resultOCPConfigMCV := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName+"admack", clusterName, &resultOCPConfigMCV, ocpConfigGetErr); err != nil {
+		return err
+	}
+
+	resultConfigMap := resultOCPConfigMCV.Status.Result
+	configMap := map[string]interface{}{}
+
+	if resultConfigMap.Raw != nil {
+		err := json.Unmarshal(resultConfigMap.Raw, &configMap)
+		utils.CheckError(err)
+	} else {
+		return ocpConfigGetErr
+	}
+
+	// Get clusterversion from managed cluster to initiate upgrade
+	managedclusterview := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+			Labels: map[string]string{
+				MCVUpgradeLabel: clusterName,
+			},
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Group:     "config.openshift.io",
+				Kind:      "ClusterVersion",
+				Name:      "version",
+				Namespace: "",
+				Version:   "v1",
+			},
+		},
+	}
+
+	mcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &mcview); err != nil {
+
+		klog.V(2).Info("Create managedclusterview " + clusterName)
+		if err := client.Create(context.TODO(), managedclusterview); err != nil {
+			return err
+		}
+	}
+
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName, clusterName, &resultmcview, getErr); err != nil {
+		return err
+	}
+
+	resultClusterVersion := resultmcview.Status.Result
+	clusterVersion := map[string]interface{}{}
```

Comment:
can we use the hive(?) api clusterversion type here or is there some problem?

Reply:
Nope, this clusterversion is from Openshift not hive. I checked hive, it doesn't seem to have one.